src/Proxy.hs

```diff
@@ -716,34 +825,32 @@ getTransactionCostR = withExchangeRate $ \(rate, pv) -> do
                     pSize
                     numSignatures
               tty' -> respond400Error (EMParseError $ "Could not parse transaction type: " <> tty') RequestInvalid
-      fetchUpdates :: ClientMonad IO (GRPCResult (EnergyRate, ProtocolVersion))
+      fetchUpdates :: ClientMonad IO (GRPCResult (Either String (EnergyRate, ProtocolVersion)))
       fetchUpdates = do
-          getConsensusStatus >>= \case
-            Left err -> return $ Left err
-            Right (GRPCResponse hds cs) -> case parseEither (withObject "Best block hash" $ \v -> v .: "bestBlock") cs of
-              Left err -> return $ Left err
-              Right bbh -> do
-                summary <- getBlockSummary bbh
-                -- This extraction of the parameter is not ideal for two reasons
-                -- - this is the exchange rate in the best block, which could already be obsolete.
-                --   This is not likely not matter since block times 10s on average, and it is always the case
-                --   that the transaction is committed after the current time. In any case this is only an estimate.
-                -- - It is manually parsing the return value, instead of using the Updates type. This should be fixed
-                --   and we want to use the same calculation in concordium-client, however that requires more restructuring
-                --   of the dependencies. The current solution is good enough in the meantime.
-                let energyRateParser = AE.withObject "Block summary" $ \obj -> do
-                      updates <- obj AE..: "updates"
-                      chainParameters <- updates AE..: "chainParameters"
-                      euroPerEnergy <- chainParameters AE..: "euroPerEnergy"
-                      microGTUPerEuro <- chainParameters AE..: "microGTUPerEuro"
-                      return $ computeEnergyRate microGTUPerEuro euroPerEnergy
-                return $ do
-                      rate <- parseEither energyRateParser . grpcResponseVal =<< summary
-                      -- the old node that supported only P1 version did not return the protocol version in consensus status.
-                      pv <- parseEither (withObject "Protocol version" $ \v -> (v AE..:! "protocolVersion" AE..!= P1)) cs
-                      return $ GRPCResponse hds (rate, pv)
+        consensusInfoRes <- getConsensusInfo 
+        case getResponseValueAndHeaders consensusInfoRes of
+            Left errRes -> return errRes
+            Right (csRes, _) -> do
+              chainParamsRes <- getBlockChainParameters Best
```

Comment:
You already have the "best" block from consensus info. Please use that here  for consistency.